### PR TITLE
summary review allows linking to specific periods

### PR DESCRIPTION
**Pull request by pandafulmanda**
Also, only scroll to section on mount when explicitly stated in route

In preparation for when we need to link to a specific period for the review page

screen shot 2015-06-22 at 1 07 10 pm

### CODE REVIEW

File: src/components/task-plan/reading-stats.cjsx [hunk 1]
```diff
@@ -194,7 +198,8 @@ Stats = React.createClass
     <BS.Panel className='reading-stats'>
       <CoursePeriodsNavShell
         handleSelect={@handlePeriodSelect}
-        intialActive={@state.period}
+        handleKeyUpdate={@props.handlePeriodKeyUpdate}
```
Comment (philschatz) [inline src/components/task-plan/reading-stats.cjsx]: Should `handlePeriodKeyUpdate` be added to `propTypes`?

**Comment by philschatz** [comment]
When changing periods should the URL update (unless it is a bunch of work)?

**Comment by pandafulmanda** [comment]
that's what this PR helps with/yes it does :smile:

**Comment by philschatz** [comment]
I did not notice the period updating when toggling between periods using the right-hand-side tabs in the little stats panel.

**Comment by pandafulmanda** [comment]
oh hmm...interesting...are you getting an errors?

**Comment by pandafulmanda** [comment]
@philschatz pushed in an update that merges the API updates, perhaps this will help with the route not updating?

**Comment by philschatz** [comment]
Maybe I'm doing something wrong but clicking the tabs in the following screenshot does not update the URL:

image

**Comment by philschatz** [comment]
can you merge from master?

**Comment by pandafulmanda** [comment]
@philschatz merged with master.  did it ever work on your computer?

**Comment by pandafulmanda** [comment]
@philschatz travis is happy, ready for review

File: src/components/task-teacher-review/index.cjsx [hunk 2]
```diff
@@ -55,11 +63,23 @@ TaskTeacherReview = React.createClass
       params.sectionIndex = stepKey + 1
       params.id = @props.id # if we were rendered directly, the router might not have the id
 
+      params.periodIndex ?= 1
```
Comment (philschatz) [inline src/components/task-teacher-review/index.cjsx]: How safe is it to update the Router params object directly?

Comment (pandafulmanda) [inline src/components/task-teacher-review/index.cjsx]: unsure, but saw this elsewhere in the code.  I'll clone it just to be safe

**Comment by pandafulmanda** [comment]
@philschatz made some updates that should add extra safety around the params situation

**Pull request merged**